### PR TITLE
Prep for RN v0.65 upgrade

### PR DESCRIPTION
**Pull request by chrisbobbe**
Related: #5230

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -24,11 +24,10 @@ buildscript {
     repositories {
         google()
         mavenCentral()
-        jcenter()
     }
     dependencies {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath 'com.android.tools.build:gradle:4.1.0'
+        classpath 'com.android.tools.build:gradle:4.2.1'
```
Comment (gnprice) [inline android/build.gradle:30]: android build: Upgrade Android Gradle Plugin to v4.2.1
    
    Done to follow the template-app change in
    facebook/react-native@7599593b3, on the path to the RN v0.65
    upgrade.
    
    Release notes here:
      https://developer.android.com/studio/releases/gradle-plugin#4-2-0
    
    Quite a few changes announced; hopefully they're all good ones. Our
    tests pass, and a debug build on Android built and ran fine for me.

Cool. I'm looking through those release notes. This one is interesting:

> Starting in AGP 4.2, it is no longer possible to override Gradle properties from subprojects. In other words, if you declare a property in a gradle.properties file in a subproject instead of the root project, it will be ignored.

We have a fair number of such files supplied by different third-party libraries. You can see all of them (plus a few that don't apply, in template or example apps rather than live subprojects) like so:
```
$ find -name gradle.properties | xargs rg -v '^ *($|#)'
```

E.g.,
```
./node_modules/react-native-webview/android/gradle.properties
1:ReactNativeWebView_kotlinVersion=1.3.50
2:ReactNativeWebView_webkitVersion=1.4.0
3:ReactNativeWebView_compileSdkVersion=29
4:ReactNativeWebView_buildToolsVersion=29.0.3
5:ReactNativeWebView_targetSdkVersion=28
6:ReactNativeWebView_minSdkVersion=21
```

Glad to not have any of those applying ad hoc anymore!

I think that's the only one that affects us.

File: package.json [hunk 2]
```diff
@@ -52,6 +52,7 @@
     "react": "17.0.1",
     "react-intl": "5.24.6",
     "react-native": "0.64.3",
+    "react-native-codegen": "^0.0.7",
```
Comment (gnprice) [inline package.json:55]: The RN repo went back and forth between having and not having this
    dependency in the template app, and I found it hard to follow why or
    what it does. But the RN v0.65 release was made with the dep
    included at this version range, so, might as well?

Following your handy links to those commits, it looks like when it got removed from the template dependencies, it was because it got moved inside the `react-native` NPM package itself.

Thinking about that, I think it means the ideal thing is probably to hold back this change to do in the same commit with the RN upgrade itself. At our current version, this isn't in the (template) package.json because it's closely coupled with the code in react-native itself.

I'm not sure whether adding it to our package.json here would do nothing (and RN would keep on invoking its own internal version of react-native-codegen), or would result in using the new react-native-codegen version from NPM. If the former, then adding it here is potentially confusing because it looks like it does something but doesn't actually. If the latter, then in principle it's incompatible -- our current RN version was explicitly meant to use its internal copy of react-native-codegen, not a separately-released one from NPM.

**Review by gnprice — COMMENTED** [review]
Thanks @chrisbobbe for your careful work on this! All looks good, except I think the last commit's changes are best held back for the main upgrade commit -- see below.

File: package.json (hunk 2 above)
Comment (chrisbobbe) [reply package.json:55]: Cool, yeah, makes sense.

> Following your handy links to those commits

For posterity, those commits were:

  facebook/react-native@58c80d4f8
  facebook/react-native@e99b8bbb4
  facebook/react-native@cd6c9f327
  facebook/react-native@98e173445
  facebook/react-native@e77595784

I think I'll make the main upgrade commit less verbose by omitting that list and linking here.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed, with those changes and an additional commit addressing some linked CZO discussion today.

#5330 is also in flight toward this RN upgrade. 🙂

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

**Comment by chrisbobbe** [comment]
Thanks for the review!
